test/release/examples/primers/LinearAlgebralib.chpl

```diff
@@ -0,0 +1,611 @@
+  the :mod:`LayoutCS` layout module.
+
+  For example, the following CSR matrix (``CS(compressRows=true)``)
+  **is supported** by this submodule:
+*/
+
+{
+  use LayoutCS;
+  const parentDom = {1..100, 1..100};
+  var csrDom: sparse subdomain(parentDom) dmapped CS();
+  var csrMatrix: [csrDom] real; // Supported by LinearAlgebra.Sparse
+}
+
+/*
+   And as a counter-example, the following COO matrix (``DefaultSparse``)
+   **is not supported** by this module:
+*/
+
+{
+  const parentDom = {1..100, 1..100};
+  var cooDom: sparse subdomain(parentDom);
+  var cooMatrix: [cooDom] real; // NOT supported by LinearAlgebra.Sparse
+}
+
+/*
+  For more information about working with sparse arrays in Chapel, see the
+  :ref:`Sparse Primer <primers-sparse>`.
+*/
+
+// Cr
+
+/*
+  Factory Functions
+  ~~~~~~~~~~~~~~~~~
+
+  The :mod:`LinearAlgebra.Sparse` submodule provides
+  factory functions for both sparse domains and sparse matrices.
+
+  In Chapel, sparse domains can only be modified directly, so it is important
+  to maintain access to both
+  the sparse array and sparse domain.
+
+  Consider the following example:
+
+*/
+{
+  // Bad example of LinearAlgebra.Sparse usage:
+  var A = CSRMatrix({1..100, 1..100}); // Create a CSR matrix from dense domain
+  writeln(A.domain); // empty sparse domain, with no way to add elements..
+}
+
+/*
+  The above sparse matrix is initialized with an empty domain, and the user
+  has no way of accessing that domain. Therefore, it is not possible to add
+  indices to the sparse domain
+
+  Instead, users should interact with both the sparse domain and sparse array:
+*/
+{
+  // Good example of LinearAlgebra.Sparse usage:
+  var D = CSRDomain(100, 100); // empty sparse domain
+  var A = CSRMatrix(D); // Create a CSR matrix from a sparse domain
+  D += [(10,20), (30, 24), (64, 42)]; // Add some indices to the sparse domain
+}
+
+/*
+  Below are a list of the available factory functions:
+*/
+
+{ // Factory functions scope
+// Creating CSR domains & arrays in Chapel without ``LinearAlgebra.Sparse``:
+use LayoutCS;
+var parentDom = {0..#3, 0..#3}; // Parent domain
+var csrDom: sparse subdomain(parentDom) dmapped CS(); // CSR domain
+var csrArr: [csrDom] real; // CSR array
+
+// Adding indices to a sparse domain:
+csrDom += (0,0);
+csrDom += [(1,1), (2,2)];
+
+// Creating CSR domains with ``LinearAlgebra.Sparse``:
+var D1 = CSRDomain(3);                 // from a length
+var D2 = CSRDomain(3, 3);              // from a length and width
+var D3 = CSRDomain(0..#3);             // from 1 range
+var D4 = CSRDomain(0..#3, 0..#3);      // from 2 ranges
+var D5 = CSRDomain({0..#3, 0..#3});    // from a 2D domain
+
+// Creating CSR matrices:
+var M1 = CSRMatrix(D1);             // from a CSR domain
+var M2 = CSRMatrix({0..#3, 0..#3}); // from a dense domain
+var M3 = CSRMatrix(M1);             // From a CSR matrix
+const I = eye(3,3);
+var M4 = CSRMatrix(I);              // From a dense matrix
```

Comment:
Is it worth stating here that only the indices corresponding to non-zero values are added to the sparse domain, or is it reasonable to expect users to look at the full docs? Up to you.